### PR TITLE
improve test coverage

### PR DESCRIPTION
**Pull request by hanars**
After all the code cleanup I have been doing I wanted to remove all the orphaned code that got accidentally left behind, so I did an audit of our unit test coverage to prune dead code and/or add test coverage

## Pull request overview

This PR primarily expands and adjusts automated test coverage across several APIs and management commands, while also refactoring multiple utility paths (pedigree parsing, Airtable patching, ClickHouse search formatting, and reference-data parsing) to simplify behavior and make it easier to test.

**Changes:**
- Added/updated a broad set of Django tests (API + management commands) to cover additional edge cases and logging behavior.
- Refactored pedigree parsing, Airtable update flows, and parts of ClickHouse search result formatting/filtering.
- Updated reference-data ingestion helpers to centralize per-model record parsing via `RECORD_FIELDS`.

### CODE REVIEW

File: clickhouse_search/managers.py [hunk 1]
```diff
@@ -1478,8 +1459,6 @@ def _filter_locations_q(self, intervals, genes=None, require_gene_filter=False):
                 interval_q |= self._interval_query(**interval)
             if locus_q is None:
                 locus_q = interval_q
-            elif require_gene_filter:
```
Comment (hanars) [inline clickhouse_search/managers.py:1484]: This flag was only used in one helper function which has since been removed: https://github.com/broadinstitute/seqr/pull/4987/changes

File: seqr/views/utils/pedigree_info_utils.py [hunk 2]
```diff
@@ -173,9 +173,7 @@ def _parse_header_columns(header, allow_id_update, update_features):
         column = None
         full_key = key
         key = key.lower()
-        if full_key in JsonConstants.JSON_COLUMNS:
-            column = full_key
-        elif key == JsonConstants.FAMILY_NOTES_COLUMN.lower():
+        if key == JsonConstants.FAMILY_NOTES_COLUMN.lower():
             column = JsonConstants.FAMILY_NOTES_COLUMN
         elif key.startswith("notes"):
```
[the remark this answers is not in the document]
Comment (hanars) [reply seqr/views/utils/pedigree_info_utils.py:178]: this code was originally added to support the datstat loading, whose code was then refactored not to use this helper so this is now unreachble code and its good to deprecate it

File: seqr/views/utils/airtable_utils.py [hunk 3]
```diff
@@ -75,8 +75,10 @@ def safe_patch_records(self, record_type, record_or_filters, record_and_filters,
                 and_filters=record_and_filters,
                 page_size=max_records + 1,
             )
-            if not records or len(records) > max_records:
-                error = f'''Unable to identify Airtable "{record_type}" record to update
+        except Exception:
+            records = []
+        if not records or len(records) > max_records:
+            error = f'''Unable to identify Airtable "{record_type}" record to update
```
[the remark this answers is not in the document]
Comment (hanars) [reply seqr/views/utils/airtable_utils.py:82]: Its irrelevant to the end user what the failure mode is here, we just want to alert the seqr engineers that they need to manually edit an airtable record

**Review by bpblanken — APPROVED** [review]
